Code/GraphMol/Sgroup.cpp

```diff
@@ -51,13 +51,39 @@ bool SGroup::hasProp(const std::string &prop) const {
   return d_prop.find(prop) != d_prop.end();
 }
 
+namespace {
+  bool isSGroupIdFree(const ROMol &mol,unsigned int id){
+    auto sgroups = getSGroups(mol);
+    CHECK_INVARIANT(sgroups,"no sgroups on molecule owning an sgroup");
+	auto match_sgroup = [&](const SGROUP_SPTR &sg) { return id == sg->getId(); };
+    return std::find_if(sgroups->begin(),sgroups->end(),match_sgroup) == sgroups->end();
+  }
+
+unsigned int getNextFreeSGroupId(const ROMol &mol) {
+  std::set<unsigned int> ids;
+  CHECK_INVARIANT(getSGroups(mol),"no sgroups on molecule");
+  for (const auto &sgroup : *getSGroups(mol)) {
+    ids.insert(sgroup->getId());
+  }
+  unsigned int nexId = 1;  // smallest possible ID
```

Comment:
Sorry, my bad: ID starts at 1, but ID still be 0 (it will be for all SGroups which auto ID assignment, indeed)... and right, it would return 4, instead of 1. I will fix that one, too.

Reply:
As I said, this function is not needed, so I'll just remove it.